language/move-prover/tests/sources/functional/opaque_native.move

```diff
@@ -4,9 +4,13 @@ module 0x42::OpaqueNative {
         pragma aborts_if_is_strict = true;
 
         // Uninterpreted spec function with axiom.
-        // Current restrictions: we cannot deal with generic functions
         fun injection(v: vector<u8>): u64;
         axiom forall v1: vector<u8>, v2: vector<u8>: v1 == v2 <==> injection(v1) == injection(v2);
+
+        // Uninterpreted generic spec function with axiom.
+        fun generic_injection<T>(x: T): u64;
+        axiom<T> forall v1: T, v2: T: v1 == v2 <==> generic_injection(v1) == generic_injection(v2);
```

Comment:
A minor point, but an injection only requires this direction `<==`. The other direction (`==>`) is related to the property of equality. I don't recall the exact name now ...

Reply:
Yeah, this is actually a bijection. It's how we have it also for `hash` in the bpl. Think we can leave for the test like this.